ouroboros-network-api/src/Ouroboros/Network/PeerSelection/RelayAccessPoint.hs

```diff
@@ -30,32 +32,66 @@ import Cardano.Binary qualified as Codec
 import Network.DNS qualified as DNS
 import Network.Socket qualified as Socket
 
+-- | Types of domains supported
+-- NB: A deliberately limited subset of SRV is supported.
+-- Concretely, a peer from only the top priority level may
+-- be given a chance to connect by the peer selection governor. Other
+-- priority levels will not be considered. If there are multiple records
+-- of the top priority (ie. lowest numerical value), a weighted random
+-- sampling is in fact performed by this implementation, and addresses
+-- and port are obtained from the winner.
+-- (cf. https://www.ietf.org/rfc/rfc2782.txt)
+--
+data DomainAccessPoint = DomainAccessPoint !DomainPlainAccessPoint
+                       -- ^ An @A@ or @AAAA@ DNS record
+                       | DomainSRVAccessPoint !DomainSRVAccessPoint
+                       -- ^ A @SRV@ DNS record
```

Comment:
Fair point - I think this was useful for my first attempt at DNSLookupResult if memory serves me correct, but I've changed the approach to the latter at some point and now this type is not strictly needed. It may still be useful for test case generation, we'll see when I'm done.